src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1698,7 +1699,9 @@ private static Replica findSuitableReplica(String keyspaceName, DecoratedKey key
         EndpointsForToken replicas = replicationStrategy.getNaturalReplicasForToken(key);
 
         // CASSANDRA-13043: filter out those endpoints not accepting clients yet, maybe because still bootstrapping
-        replicas = replicas.filter(replica -> StorageService.instance.isRpcReady(replica.endpoint()));
+        // We have a keyspace, so filter by affinity too
+        replicas = replicas.filter(replica -> StorageService.instance.isRpcReady(replica.endpoint()))
+                           .filter(snitch.filterByAffinity(keyspace.getName()));
```

Comment:
This change corresponds to the DSE change in `CounterWriteHandlerSupplier` which was moved unchanged from `StorageProxy`.